sources/packages/backend/apps/api/src/services/application/application.service.ts

```diff
@@ -80,23 +92,25 @@ export class ApplicationService extends RecordDataModelService<Application> {
     private readonly offeringService: EducationProgramOfferingService,
     private readonly notificationActionsService: NotificationActionsService,
     private readonly institutionLocationService: InstitutionLocationService,
+    private readonly notificationActionService: NotificationActionsService,
   ) {
     super(dataSource.getRepository(Application));
+    this.studentRepo = this.dataSource.getRepository(Student);
   }
 
   /**
    * Submits a Student Application.
    * If the application is already in Draft state, then on submit the, existing
    * row will be updated with the payload and application status will be set to
    * Submitted and applicationStatusUpdatedOn will also be update and new workflow is started.
-   * If a student submit/re-submit and an existing application that is not in draft state
+   * If a student submit/re-submit an existing application that is not in draft state
    * (i.e existing application should be in any one of these status, submitted, In Progress,
    * Assessment, Enrollment), then the existing application status is set to `Overwritten` and
    * applicationStatusUpdatedOn is updated and delete the corresponding workflow and creates a
    * new Application with same Application Number and Program Year as that of the Overwritten
    * Application and with newly submitted payload. And starts a new workflow for the newly created
-   * Application.
-   * a new application with status 'Submitted'
+   * Application. If the application is edited for the 5th time, a notification is saved for the
```

Comment:
I'm not sure if it is a good idea to hardcode numbers like '5th time' in the comments because it might change.